spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/convert/AggregateReferenceConverters.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.jdbc.core.convert;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.springframework.core.ResolvableType;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.core.convert.TypeDescriptor;
+import org.springframework.core.convert.converter.GenericConverter;
+import org.springframework.data.convert.ReadingConverter;
+import org.springframework.data.convert.WritingConverter;
+import org.springframework.data.jdbc.core.mapping.AggregateReference;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+/**
+ * Converters for aggregate references. They need a {@link ConversionService} in order to delegate the conversion of the
+ * content of the {@link AggregateReference}.
+ *
+ * @author Jens Schauder
+ * @since 2.6
+ */
+final class AggregateReferenceConverters {
```

Comment:
`final` on a package-private class is unusual. Can we remove `final` here?

Reply:
Done